plugins/router/template/types.go

```diff
@@ -1,27 +1,38 @@
 package templaterouter
 
-type Frontend struct {
-	Name          string
-	HostAliases   []string
-	Backends      map[string]Backend
+import (
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+// A frontend is a representation of a service along with the corresponding backends that
```

Comment:
All Godoc should be of the form `// <type|func_name> is a`.  So `// ServiceUnit is a`.  Please do a pass on all your godoc (this includes fields).